database/mongo/controller.mjs

```diff
@@ -0,0 +1,237 @@
+                return { "endpoint_error": "updateOne", "status": 400, "message": `Cannot find 'type' on this data, and so cannot figure out a collection for it.` }
+            if (!data_type)
+                return { "endpoint_error": "updateOne", "status": 400, "message": `Cannot find 'type' on this data, and so cannot figure out a collection for it.` }
+            collection = discernCollectionFromType(data_type)
+            if (!collection)
+                return { "endpoint_error": "updateOne", "status": 500, "message": `Cannot figure which collection for object of type '${data_type}'` }
+            const obj_id = data_id.split("/").pop()
+            const filter = { "_id": data_id }
+            const result = await this.db.collection(collection).replaceOne(filter, data)
+            if (result?.matchedCount === 0) {
+                return { "endpoint_error": "updateOne", "status": 404, "message": `id '${obj_id}' Not Found` }
+            }
+            if (result?.modifiedCount >= 0) {
+                return data
+            } else {
+                return { "endpoint_error": "updateOne", "status": 500, "message": "Document was not updated in the database." }
+            }
+        } catch (err) {
+            console.error(err)
+            return { "endpoint_error": "updateOne", "status": 500, "message": "There was an error updating the document in the database." }
+        }
+    }
+
+    /**
+     * Update an existing object in the database (mongo)
+     * @param data JSON from an HTTP DELETE request.  It must contain an id.
+     * @return The delete result JSON or error JSON
+     */
+    async remove(id) {
+        return { "endpoint_error": "deleteOne", "status": 501, "message": `Not yet implemented.  Stay tuned.` }
+
+        // We have to query for the object to get the type to know the collection to remove from
+        const data_type = ""
+        let collection = null
+        if (!id)
+            return { "endpoint_error": "deleteOne", "status": 400, "message": `Cannot find 'type' on this data, and so cannot figure out a collection for it.` }
+        if (!data_type)
+            return { "endpoint_error": "deleteOne", "status": 400, "message": `Cannot find 'type' on this data, and so cannot figure out a collection for it.` }
+        collection = discernCollectionFromType(data_type)
+        if (!collection)
+            return { "endpoint_error": "deleteOne", "status": 500, "message": `Cannot figure which collection for object of type '${data_type}'` }
+
+        const result = await this.db.collection(collection).deleteOne(query, { $set: update })
+        if (result?.ok) {
+            return result
+        } else {
+            return { "endpoint_error": "deleteOne", "status": 500, "message": result.message }
+        }
+    }
+
+    /**
+     * Get by ID.  We need to decide about '@id', 'id', '_id', and http/s 
+     */
+    async getByID(id) {
+        const result = await this.query({ "_id": id })
```

Comment:
It might be useful for this to also accept an `id` and `@id` though I haven't been through the use cases in my mind yet.